chains/evm/cli/admin/withdraw.go

```diff
@@ -14,99 +26,122 @@ var withdrawCmd = &cobra.Command{
-Decimals: %v`, Amount, TokenID, Bridge, Handler, Token, Recipient, Decimals)
+func ValidateWithdrawCmdFlags(cmd *cobra.Command, args []string) error {
+	if !common.IsHexAddress(Bridge) {
+		return fmt.Errorf("invalid bridge address: %s", Bridge)
+	}
+	if !common.IsHexAddress(Handler) {
+		return fmt.Errorf("invalid handler address: %s", Handler)
+	}
+	if !common.IsHexAddress(Token) {
+		return fmt.Errorf("invalid token address: %s", Token)
+	}
+	if !common.IsHexAddress(Recipient) {
+		return fmt.Errorf("invalid recipient address: %s", Recipient)
+	}
+	if TokenID != "" && Amount != "" {
+		return errors.New("only id or amount should be set")
+	}
+	if TokenID == "" && Amount == "" {
+		return errors.New("id or amount flag should be set")
+	}
+	return nil
 }
 
-/*
+func ProcessWithdrawCmdFlags(cmd *cobra.Command, args []string) error {
+	var err error
 
-func withdraw(cctx *cli.Context) error {
-	url := cctx.String("url")
-	gasLimit := cctx.Uint64("gasLimit")
-	gasPrice := cctx.Uint64("gasPrice")
-	sender, err := cliutils.DefineSender(cctx)
-	if err != nil {
-		return err
-	}
-	bridgeAddress, err := cliutils.DefineBridgeAddress(cctx)
+	bridgeAddr = common.HexToAddress(Bridge)
+	handlerAddr = common.HexToAddress(Handler)
+	tokenAddr = common.HexToAddress(Token)
+	recipientAddr = common.HexToAddress(Recipient)
+	decimals := big.NewInt(int64(Decimals))
+	realAmount, err = calls.UserAmountToWei(Amount, decimals)
 	if err != nil {
 		return err
 	}
+	return nil
+}
 
-	handler := cctx.String("handler")
-	if !common.IsHexAddress(handler) {
-		return fmt.Errorf("invalid handler address %s", handler)
+func WithdrawCmd(cmd *cobra.Command, args []string, txFabric calls.TxFabric, gasPricer utils.GasPricerWithPostConfig) error {
+	// fetch global flag values
+	url, gasLimit, gasPrice, senderKeyPair, err := flags.GlobalFlagValues(cmd)
+	if err != nil {
+		return fmt.Errorf("could not get global flags: %v", err)
 	}
-	handlerAddress := common.HexToAddress(handler)
 
-	token := cctx.String("token")
-	if !common.IsHexAddress(token) {
-		return fmt.Errorf("invalid token address %s", token)
-	}
-	tokenAddress := common.HexToAddress(token)
+	fmt.Printf("Withdrawing %s token from handler: %s", Amount, Handler)
 
-	recipient := cctx.String("recipient")
-	if !common.IsHexAddress(recipient) {
-		return fmt.Errorf("invalid recipient address %s", recipient)
+	ethClient, err := evmclient.NewEVMClientFromParams(url, senderKeyPair.PrivateKey())
+	if err != nil {
+		log.Error().Err(fmt.Errorf("eth client initialization error: %v", err))
+		return err
 	}
-	recipientAddress := common.HexToAddress(recipient)
+	gasPricer.SetClient(ethClient)
+	gasPricer.SetOpts(&evmgaspricer.GasPricerOpts{UpperLimitFeePerGas: gasPrice})
 
-	amount := cctx.String("amount")
-	id := cctx.String("id")
+	// @dev withdrawal data should include:
+	// tokenAddr => common.Address
+	// recipientAddr => common.Address
+	// amountOrTokenID => uint256 (*big.Int)
+	withdrawalData := bytes.Buffer{}
```

Comment:
Maybe move this to `PrepareWithdrawInput` and accept `tokenAddr`, `recipientAddr` and `realAmount` as arguments so we have it properly typed.

Also, it would be nice just to have `Withdraw` function instead of calling `Transact` as we will do the #124 refactor soon and it would need less refactoring.

Reply:
going to push some changes to see if they are more in line with what you are describing